src/routes/posts.js

```diff
@@ -0,0 +1,35 @@
+const express = require('express')
+
+const db = require('../db')
+
+const router = express.Router()
+
+// CREATE AN ARTICLE
+
+const createPost = async (req, res, next) => {
+    try {
+        const { id, userId, title, image, content, published, createdAt } =
+            req.body
+        await db.query(
```

Comment:
> also, how do i send id and createdAt in the response body if I don't get it from the request body ? @jocrah

```suggestion
       const newPost = await db.query(
```

@beblicarl, by assigning the insert to a variable and using that. Don't really know the response that will be in the variable but ideally the info should be there

Reply:
so it would be newPost.createdAt and newPost.id ?

Reply:
correct.  kindly do check what the command returns and lets see if it has the data. you can share it here if possible

Reply:
![image](https://user-images.githubusercontent.com/68056104/185174264-4a1d9a9a-d505-4ebd-a751-d9a28322c8e8.png)

I am getting this error, newArticle is never reassigned

Reply:
cool cool, making progress. it means we need to do a bit more work then.

the `id` column type must be updated in the migrations to autogenerate on its own in the database then.

for sure we can't be taking ids from the request for a primary key field because it means the frontend always has to know what key to send that will not conflict with the existing keys in the db

found this so far so do take a look at it and let me know any ideas you come up with: https://chartio.com/resources/tutorials/how-to-define-an-auto-increment-primary-key-in-postgresql/

what we want for now is to create a new migration and modify the id column so that it autogenerates on its own. I'll join you on that later but do let me know what you find :wink:

Reply:
that's perfect !
How about the createdAt as well ?

Reply:
i don't understand what you mean with the current_timestamp

Reply:
cool. glad you're asking questions :smile:. so i meant the existing migration should handle that, if you see line 9. So you can just try passing everything else and exclude the `createdAt` and see if the insert works

https://github.com/tbf-challenge/teamwork-be/blob/b4a4f5817504c2590f509112e27514c240e2fc14/migrations/1660111562812_db-init.js#L6-L10

Reply:
for more context, the `current_timestamp` is a key word in most SQL definitions to say it should use the date and time at that particular instant as the default.

https://www.postgresqltutorial.com/postgresql-date-functions/postgresql-current_timestamp/

Reply:
oh okay, i carefully read through it.. so how i understand it like you said early on, the id as long as the current_timestamp should be handled in the migration

Reply:
maybe we're saying the same thing but i mean for the `createdAt`, the default being `current_timestamp` means it will always be generated correctly for you if you insert a document. You don't have to create your own date and pass it along

Reply:
yeah, i meant the same thing. What I mean is, during the creation of the various tables ;
1.  id takes a value of a SERIAL PRIMARY key, which means it will increment automatically anytime a new document is added
2. createdAt has a default value being current_timestamp, which means it automatically provides the current time anytime a document is added.


Just to be sure, are we on the same page?

Reply:
Exactly, yes we are on the same page

Reply:
awesome. I guess i'd wait till later for you to join me with the other things

Reply:
@beblicarl seems @jakazzy was also experiencing an id issue and has worked on the [fix](https://github.com/tbf-challenge/teamwork-be/pull/75) for that. so will merge that so you can pull those changes into yours and see how it goes.

You can ask her the reason for the changes she made to learn more. Kindly follow the manual steps in the pr so everything works smoothly

After that. we can go again on this and see how it works out